projects/igniteui-angular/src/lib/select/select.component.ts

```diff
@@ -312,9 +377,51 @@ export class IgxSelectComponent extends IgxDropDownComponent implements IgxSelec
 
     /** @hidden @internal */
     public onBlur(): void {
+        if (this.ngControl && !this.ngControl.valid) {
+            this.valid = IgxSelectState.INVALID;
+        } else {
+            this.valid = IgxSelectState.INITIAL;
+        }
+
         if (!this.collapsed) {
             this.toggleDirective.close();
         }
     }
+
+    protected onStatusChanged() {
+        if ((this.ngControl.control.touched || this.ngControl.control.dirty) &&
+            (this.ngControl.control.validator || this.ngControl.control.asyncValidator)) {
+            this.valid = this.ngControl.valid ? IgxSelectState.VALID : IgxSelectState.INVALID;
+        }
+    }
+    /**
+     * @hidden @internal
+     */
+    public ngOnInit() {
+        try {
+            this.ngControl = this._injector.get(NgControl);
+        } catch (e) { }
```

Comment:
Is wrapping this in a `try - catch` necessary? I think `this._injector` will always be provided in an Angular app

Reply:
The try - catch is not because of the injector, but because of the exception, thrown from the [get()](https://angular.io/api/core/Injector#get) method, when ngControl cannot be found.